resources/dw/resource_aws_cluster.go

```diff
@@ -0,0 +1,137 @@
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dw
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+)
+
+type dwAwsClusterResource struct {
+	client *cdp.Client
+}
+
+var (
+	_ resource.Resource              = (*dwAwsClusterResource)(nil)
+	_ resource.ResourceWithConfigure = (*dwAwsClusterResource)(nil)
+)
+
+func NewAwsClusterResource() resource.Resource {
+	return &dwAwsClusterResource{}
+}
+
+func (r *dwAwsClusterResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *dwAwsClusterResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dw_aws_cluster"
+}
+
+func (r *dwAwsClusterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = dwAwsClusterSchema
+}
+
+func (r *dwAwsClusterResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var plan awsClusterResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Generate API request body from plan
+	clusterParams := operations.NewCreateAwsClusterParamsWithContext(ctx).
+		WithInput(plan.convertToCreateAwsClusterRequest())
+
+	// Create new aws cluster
+	response, err := r.client.Dw.Operations.CreateAwsCluster(clusterParams)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating data warehouse cluster",
+			"Could not create cluster, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	payload := response.GetPayload()
+	desc := operations.NewDescribeClusterParamsWithContext(ctx).
+		WithInput(&models.DescribeClusterRequest{ClusterID: &payload.ClusterID})
+	describe, err := r.client.Dw.Operations.DescribeCluster(desc)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating data warehouse cluster",
+			"Could not describe cluster, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	cluster := describe.GetPayload()
+
+	// Map response body to schema and populate Computed attribute values
+	plan.ID = types.StringValue(cluster.Cluster.EnvironmentCrn)
+	plan.Crn = types.StringValue(cluster.Cluster.EnvironmentCrn)
```

Comment:
either `id`/`crn`  is sufficient

Reply:
for some reason ID is needed for terraform, the CRN is ours